helpers/cache.js

```diff
@@ -46,11 +46,12 @@ module.exports.connect = function(cacheEnabled, config, logger, cb) {
 
 	client.on('ready', () => {
```

Comment:
So the code works this way (which is not straightforward to understand and doesn't seem to be correct):

Try to connect to Redis, return null the first time though. Call callback if the connection was successful. For every error since now on, log the error. If the connection to Redis was unsuccessful, the connection attempt will be retried every 10 seconds by default and error will be logged each time. 

I suggest changing the connection behaviour, so the `connect` function is being called just once and the connection to Redis is being closed in case of an error. 
```suggestion
      return new Promise((resolve, reject) => {
		client.on('ready', resolve);
		client.on('error', reject);
	}).then(() => {
		// Called after "ready" Redis event
		logger.info('App connected with redis server');
	}).catch(err => {
		// Called after "error" Redis event
		logger.info('Redis:', err);
		// Do not attempt to reconnect after an error
		client.quit();
	}).finally(() =>
		// Redis usage is optional; return successful message regardless of the Redis connect attempt outcome
		cb(null, { cacheEnabled, client })
	);
```

Reply:
@diego-G I pushed the changes directly (https://github.com/LiskHQ/lisk/pull/2639/commits/436a6c174948c9762a7fe5e53e670f500c76ce64), the explanation would be too prolix. Please follow the comments and let me know what do you think.